lib/collection/src/operations/verification/query.rs

```diff
@@ -35,6 +35,52 @@ impl Query {
         }
         Ok(())
     }
+
+    /// Check that the query does not perform a fullscan based on the collection configuration.
+    async fn check_fullscan(
+        &self,
+        using: &str,
+        collection: &Collection,
+        strict_mode_config: &StrictModeConfig,
+    ) -> CollectionResult<()> {
+        // Check only applies on `unindexed_filtering_retrieve`
+        if strict_mode_config.unindexed_filtering_retrieve == Some(false) {
```

Comment:
Nit but maybe we can early return here to reduce amount of indents.

Reply:
I prefer keeping it this to make the scope clearer when add new checks in the custom checker.